--- Portfile1	2015-08-03 12:42:32.000000000 -0400
+++ Portfile	2015-08-03 12:47:24.000000000 -0400
@@ -1,12 +1,13 @@
 # $Id: Portfile 114431 2013-12-08 09:48:42Z and.damore@macports.org $
+
 PortSystem 1.0
 
 name                vnstat
-version             1.11
+version             1.14
 revision            2
 categories          net
 platforms           darwin
-maintainers         nomaintainer
+maintainers         gmail.com:youvegotmoxie
 license             GPL-2
 description         console-based network traffic monitor
 long_description    \
@@ -14,14 +15,12 @@
 	a log of network traffic for the selected interface(s), it uses the network \
 	interface statistics provided by the kernel as information source.  This means \
 	that vnStat won't actually be sniffing any traffic and also ensures light use \
-	of system resources. 
+	of system resources.
 
 homepage            http://humdi.net/vnstat/
-depends_lib         port:gd2       
+depends_lib         port:gd2
 master_sites        http://humdi.net/${name}
-checksums           md5     a5a113f9176cd61fb954f2ba297f5fdb \
-                    sha1    92494f38a752dcf60053af2a6d8977737da7e616 \
-                    rmd160  c7a8602c6ed86c674b14b2b8708e97f5999fb0e5
+checksums           sha256    f8462a47d85d0890493dc9eaeafbc725ae631aa5b103fb7f8af4ddb2314e8386
 
 post-patch {
     # fix Makefile
@@ -29,7 +28,7 @@
     reinplace "s|${prefix}/man|${prefix}/share/man|g"    ${worksrcpath}/Makefile
     reinplace "s|(DESTDIR)/etc|(DESTDIR)${prefix}/etc|g" ${worksrcpath}/Makefile
     reinplace "s|(DESTDIR)/var|(DESTDIR)${prefix}/var|g" ${worksrcpath}/Makefile
-    
+
     # fix occurrences of /var/db references in other files
     reinplace "s|/var|${prefix}/var|g"  ${worksrcpath}/src/common.h \
                                         ${worksrcpath}/cfg/vnstat.conf \
@@ -39,7 +38,7 @@
     reinplace "s|/var/lib|/var/db|g"    ${worksrcpath}/man/vnstat.1 \
                                         ${worksrcpath}/man/vnstatd.1 \
                                         ${worksrcpath}/man/vnstati.1
-    
+
     # fix occurrences of /etc/db references in other files
     reinplace "s|/etc|${prefix}/etc|g"  ${worksrcpath}/man/vnstat.1 \
                                         ${worksrcpath}/man/vnstat.conf.5 \
@@ -61,12 +60,27 @@
 post-destroot {
     xinstall -m 755 -d ${destroot}${prefix}/share/doc/${name}
     xinstall -m 755 -d ${destroot}${prefix}/share/examples/${name}
-    
+
     xinstall -m 644 -W ${worksrcpath} \
         CHANGES COPYING FAQ INSTALL INSTALL_BSD README \
         ${destroot}${prefix}/share/doc/${name}
 
     eval file copy [glob ${worksrcpath}/examples/*] \
         ${destroot}${prefix}/share/examples/${name}
+	 
+	 # handle old config
+	 file rename "${destroot}${prefix}/etc/vnstat.conf" "${destroot}${prefix}/etc/vnstat.conf-local"
 }
 
+# handle old config
+post-activate {
+    if {![file exists "${prefix}/etc/vnstat.conf"]} {
+        copy "${prefix}/etc/vnstat.conf" "${prefix}/etc/vnstat.conf-local"
+    }
+}
+
+notes "
+
+If not existing, a default vnstat.conf has been created in ${prefix}/etc/vnstat.conf-local
+Copy to ${prefix}/etc/vnstat.conf to get started.
+"
